src/iso9660.rs

```diff
@@ -567,8 +579,23 @@ mod tests {
         IsoFs::from_file(iso_file).unwrap()
     }
 
+    fn open_truncated_iso() -> Result<IsoFs, Error> {
+        let iso_bytes: &[u8] = include_bytes!("../fixtures/iso/synthetic.iso.xz");
+        let mut decoder = XzDecoder::new(iso_bytes);
+        let mut iso_file = tempfile().unwrap();
+        copy(&mut decoder, &mut iso_file).unwrap();
+        iso_file
+            .set_len(iso_file.metadata().unwrap().len() / 2)
+            .unwrap();
+        IsoFs::from_file(iso_file)
+    }
+
     #[test]
     fn test_primary_volume_descriptor() {
+        assert_eq!(
+            open_truncated_iso().unwrap_err().to_string(),
+            "ISO image is incomplete"
+        );
         let iso = open_iso();
         let desc = iso.get_primary_volume_descriptor().unwrap();
         assert_eq!(desc.system_id, "system-ID-string");
```

Comment:
We should assert the value of `volume_space_size` also.  `isoinfo -d` says it should be 338.  I think it's okay to hardcode assumptions about the test fixture; we can always update the test if it changes.